managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1956,7 +1960,7 @@ boolean ledgerExists(long ledgerId) {
         return ledgers.get(ledgerId) != null;
     }
 
-    long getNextValidLedger(long ledgerId) {
+    Long getNextValidLedger(long ledgerId) {
```

Comment:
Why changing type? Do we need null?

Reply:
yes, because `long` was causing NPE if we try to get next ledger after very last ledger in the list. at that time `ledgers.ceilingKey(ledgerId + 1)`  returns null and it gives NPE if we try to cast it in long so, changed the data type to Long and handling at caller.